src/ranged.cpp

```diff
@@ -1358,6 +1358,9 @@ void make_gun_sound_effect(player &p, bool burst, item *weapon)
         sounds::sound(p.pos(), 4, _("Fwoosh!"));
     } else if( ammo_used != "bolt" && ammo_used != "arrow" && ammo_used != "pebble" &&
                ammo_used != "fishspear" && ammo_used != "dart" ) {
+               if (noise < 0) {
```

Comment:
You could invert the `if` and integrate it into the above. ie.
```
if( noise > 0 && ammo_used !=... ) {
    sound::sound
}
```

That way 0 noise noises without description wouldn't be produced.